src/tools/request-merge.js

```diff
@@ -4,6 +4,51 @@ const _ = require('lodash');
 
 const requestClean = require('./request-clean');
 
+// Do a merge with case-insensitive keys, and drop empty header keys
+const caseInsensitiveMerge = (requestOne, requestTwo, requestThree) => {
+  // This creates copies/clones
+  requestOne = requestClean(requestOne);
+  requestTwo = requestClean(requestTwo);
+  requestThree = requestClean(requestThree);
+
+  // This is a very quick & efficient merge for all of request's properties
+  const mergedRequest = _.merge.apply(_, [requestOne, requestTwo, requestThree]);
+
+  // Now to cleanup headers, we start on the last request (the one to keep) and work backwards to add the keys
+  // NOTE: This is done "manually" instead of a _.merge or Object.assign() because we need case-insensitivity
+  const mergedRequestHeaders = requestThree.headers || {};
+  const requestTwoHeaders = requestTwo.headers || {};
+  const requestOneHeaders = requestOne.headers || {};
+
+  // Check, in order, which keys to add (if they're not duplicate)
+  [requestTwoHeaders, requestOneHeaders].forEach((requestHeaders) => {
+    const existingKeys = Object.keys(mergedRequestHeaders);
+    const requestKeys = Object.keys(requestHeaders);
+
+    // We will loop through every header, and if we find no (case-insensitive) match, we'll add it to mergedRequest
+    requestKeys.forEach((checkingKey) => {
```

Comment:
Great point. Do you think it'll improve readability or performance considerably? I'm thinking the header object will never have _many_ keys, but I might be missing something.